ee/src/test/java/org/jboss/as/ee/subsystem/EeOperationsTestCase.java

```diff
@@ -206,6 +210,32 @@ public void testAddExistingGlobalModuleWithMetaInf() throws Exception {
         assertTrue(actualModule.get(GlobalModulesDefinition.META_INF).asBoolean());
     }
 
+    @Test
+    public void testAddExistingComponent(){
+        EEModuleDescription eeModuleDescription = new EEModuleDescription("appName","module1",
+                "ear1",false);
+        ComponentDescription description1 = new ComponentDescription("comp1",
+                "org.test.comp1",
+                eeModuleDescription,
+                ServiceName.of("name"));
+        ComponentDescription description2 = new ComponentDescription("comp1",
+                "org.test.comp2",
+                eeModuleDescription,
+                ServiceName.of("name"));
+
+        eeModuleDescription.addComponent(description1);
+        Exception exception = assertThrows(RuntimeException.class, () -> {
+            eeModuleDescription.addComponent(description2);
+        });
+
+        String expectedMessage = "WFLYEE0040: Component 'comp1' in class 'org.test.comp2' " +
+                "is already defined in class 'org.test.comp1'";
+        String actualMessage = exception.getMessage();
+        assertTrue(actualMessage.equalsIgnoreCase(expectedMessage));
```

Comment:
Please have the test confirm the message contains "WFLYEE0040", "'comp1'", "'org.test.comp2'" and "'org.test.comp1'" instead of checking the entire string. The exception message is internationalized and checking the full English language text of messages has proven to be a problem in the past when tests get run in environments with localized text.